main.go

```diff
@@ -0,0 +1,517 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package main implements HarbourBridge, a stand-alone tool for Cloud Spanner
+// evaluation, using data from an existing PostgreSQL database. See README.md
+// for details.
+package main
+
+import (
+	"bufio"
+	"context"
+	"crypto/rand"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+	"sync/atomic"
+	"syscall"
+	"time"
+
+	sp "cloud.google.com/go/spanner"
+	database "cloud.google.com/go/spanner/admin/database/apiv1"
+	instance "cloud.google.com/go/spanner/admin/instance/apiv1"
+	"google.golang.org/api/iterator"
+	adminpb "google.golang.org/genproto/googleapis/spanner/admin/database/v1"
+	instancepb "google.golang.org/genproto/googleapis/spanner/admin/instance/v1"
+
+	"harbourbridge/internal"
+	"harbourbridge/internal/postgres"
+	"harbourbridge/spanner"
+	"harbourbridge/spanner/ddl"
+)
+
+var (
+	badDataFile      = "dropped.txt"
+	schemaFile       = "schema.txt"
+	reportFile       = "report.txt"
+	dbNameOverride   string
+	instanceOverride string
+	filePrefix       = ""
+	verbose          bool
+)
+
+func main() {
+	flag.Usage = usage
+	flag.Parse()
+	internal.VerboseInit(verbose)
+	now := time.Now()
+	lf, err := setupLogFile()
+	if err != nil {
+		fmt.Printf("\nCan't set up log file: %v\n", err)
+		panic(fmt.Errorf("can't set up log file"))
+	}
+	defer close(lf)
+	project, err := getProject()
+	if err != nil {
+		fmt.Printf("\nCan't get project: %v\n", err)
+		panic(fmt.Errorf("can't get project"))
+	}
+	instance, err := getInstance(project)
+	if err != nil {
+		fmt.Printf("\nCan't get instance: %v\n", err)
+		panic(fmt.Errorf("can't get instance"))
+	}
+	fmt.Printf("WARNING: Please check that permissions for this instance are appropriate.\n")
```

Comment:
I'm not sure about these print statements in terms of maintainability but I think the alternatives will look less tidy so if you think it's unlikely to change then I'll allow it.

Reply:
I moved them out of main and used a single raw string: easier to manage formatting.
I also tweaked it a bit: there are now newlines before and after the warning to highlight it (previously it kind of got a little lost in the rest of the text).
Does that help?